File: app/src/main/java/com/vivek/wo/mvvm/sample/MainActivity.java
package com.vivek.wo.mvvm.sample;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.vivek.wo.common.dialog.BottomSheetDialog;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    public void onProgressDialog(View view) {
        BottomSheetDialog dialog = new BottomSheetDialog(this) {
            @Override
            protected View createContentView() {
                return LayoutInflater.from(getContext()).inflate(R.layout.dialog_content_tips, null, false);
            }
        };
        dialog.show();
    }

}

File: app/src/main/java/com/vivek/wo/mvvm/sample/MainActivity.kt
package com.vivek.wo.mvvm.sample

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v7.app.AppCompatActivity;
import android.view.Menu
import android.view.MenuItem
import com.vivek.wo.mvvm.sample.login.LoginActivity

import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    lateinit var mainViewModel: MainViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        fab.setOnClickListener { view ->
            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                .setAction("Action", null).show()
        }

        mainViewModel = MainViewModel()

        val mainActivityFragment = MainActivityFragment()
        mainActivityFragment.mainViewModel = mainViewModel

        supportFragmentManager.beginTransaction()
            .add(R.id.main_fragment_container, mainActivityFragment)
            .commit()

    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_login -> {
                startActivity(Intent(this@MainActivity, LoginActivity::class.java))
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}
